parser/src/lib.rs

```diff
@@ -93,12 +105,16 @@ impl From<Ast> for Element {
                 environment: HashMap::new(),
                 children: tag.elements.into_iter().map(|e| e.into()).collect(),
             },
-            Ast::Module(module) => ModuleInvocation {
-                name: module.name,
-                args: module.args,
-                body: module.body,
-                one_line: module.one_line,
-            },
+            Ast::Module(module) => {
+                match module.args {
+                    MaybeArgs::ModuleArguments(args) => ModuleInvocation {
+                        name: module.name,
+                        args,
+                        body: module.body,
+                        one_line: module.one_line,
+                    },
+                    MaybeArgs::Error(error) => panic!("{}", error),
```

Comment:
I think that this shouldn't panic. This means that if one user writes any invalid arguments, the program just crashes. Try changing the trait from `From<T>` to `TryFrom<T>` to allow for one success type and one failure type, rather than panicking!

Reply:
This would require a lot of rewriting because we can not use into on any instance of Ast but instead try_into which would require checking for Result everywhere but maybe it is worth it.

Reply:
I do not think we need to do so much rewriting. If we do TryFrom, just wrap every case you currently have into an `Ok(...)` and the panicing case into a `Err(...)`. Note that this is just the final step when converting to Element, so no `Result`s will be in our Ast. Also, note chaining errors with the question mark operator. See the example on the `TryFrom` doc page on `GreaterThanZero(i32)` for an example

Reply:
![image](https://user-images.githubusercontent.com/73667195/218500934-25339092-a15f-4d07-b07e-53192503655d.png)

```the trait bound `Element: From<Ast>` is not satisfied
required for `Ast` to implement `std::convert::Into<Element>`rustc[Click for full compiler diagnostic](rust-analyzer-diagnostics-view:/diagnostic%20message%20%5B2%5D?2#file%3A%2F%2F%2Fhome%2Fbruhng%2FGitRepos%2Fmodmark%2Fparser%2Fsrc%2Flib.rs)
lib.rs(104, 68): required by a bound introduced by this call```

This is because we do not implement into anymore but try_into. But maybe we do not use it in that many places.

Reply:
We just need to change our .into() to .try_into(), I don't think it will be a problem

Reply:
But then how do we handle try into when it fails? Do we panic? We are back where we started

Reply:
I don't know the best way to handla it either so I am open to suggestions. If the playground panics it just doesn't update untill the text is fixed, it does not crash, if that helps our decision, we can discuss it tomorrow or Thursday

Reply:
I do not think the code should panic at all. The best way, in my opinion, would be to propagate it to callee, as written in the original issue. The callee would need to handle the unwrapping and what to do if something fails, which is hopefully not panicking, but rather showing an error message or something. And yes, the HTML doesn't crash in the playground, but the panic makes the entire rust-part of the playground crash, which would lead to bad things if we would have persistant state which may not be recoverable, or loaded/cached modules which would be lost. One way would be to try to catch the panic and unwind - which would lose all diagnostics and add overhead. The other option is to use `TryFrom` instead, which I think works fine. The rust docs also says that this is the way to do it - `From` may not fail but `TryFrom` may return an error. I still strongly suggest we use that and don't push panicking code to main. If we run the program with the watcher from cli, it would actually completly crash